Source/SwiftLintFramework/Rules/Lint/CallSuperOnlyRule.swift

```diff
@@ -0,0 +1,74 @@
+import SourceKittenFramework
+
+public struct CallSuperOnlyRule: Rule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "call_super_only",
+        name: "Call Super Only",
+        description: "Methods that don't do anything but call `super` can be removed",
+        kind: .lint,
+        nonTriggeringExamples: [
+            """
+            override func viewDidDisappear(_ animated: Bool) {
+                childViewController.viewDidDisappear(animated)
+            }
+            """,
+            """
+            override func viewDidDisappear(_ animated: Bool) {
+                super.viewDidDisappear(animated)
+                print("View controller did disappear")
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            ↓override func a(){/*comment*/super.a()}
+            """,
+            """
+            ↓override func viewDidLoad() {
+                super.viewDidLoad()
+
+                // Do any additional setup after loading the view.
+            }
+            """,
+            """
+            ↓override func didReceiveMemoryWarning() {
+                super.didReceiveMemoryWarning()
+                // Dispose of any resources that can be recreated.
+            }
+            """,
+            """
+            ↓override func becomeFirstResponder() -> Bool {
+                return super.becomeFirstResponder()
+            }
+            """
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let paramsOrArguments = "(\\([\\w\\s,:_()=?\\->]*\\))"
+        let whitespaceOrComments = "(\\s|//[^\\n\\r]*|/\\*[^\\n\\r]*\\*/)*+"
+        let signature = "override[\\w,\\s]*func\\s(\\w+)\(paramsOrArguments)[\\w\\s\\->]*"
+        let body = "\\{\(whitespaceOrComments)(return\\s)?super\\.\\1\(paramsOrArguments)\(whitespaceOrComments)\\}"
+
+        let pattern = signature + body
+
+        return file
+            .match(pattern: pattern)
+            .compactMap { range, syntaxKinds in
+                // Skip matches that occur in strings and comments
+                guard syntaxKinds != [.string],
+                    syntaxKinds != [.comment]
+                    else { return nil }
```

Comment:
I'd really like a better solution for this.

If I use `.match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)`, the rule will stop working, because comments are part of this rule.

Reply:
can you try implementing this as an `ASTRule`? Using regexes is very error prone and one of the major causes of bugs and slowness in SwiftLint.

You can check for method declarations with the `override` attribute and then search for method calls inside that declaration's body.

Reply:
That sounds like a great idea 👍 

I also noticed I get a lot of false positives from the open source repos with the regex implementation.

I'll figure it out and update the PR.

Reply:
Awesome! Feel free to ask any questions 💯

Reply:
So, `ASTRule`s are awesome and the implementation is much simpler now. Thanks 🙌!

I have a problem with the offset of the expected failure location markers (`↓`).
I'm pretty sure the offset that is calculated is at the right spot, but the test fails anyway.

Can you take a look at my latest commit?

Reply:
Also, I'm getting a lot of false positives when the body contains a variable assignment (which doesn't show up in the `[String: SourceKitRepresentable]` dictionary).

Do you think it's sufficient to check for an `=` in the method body?